public/app/features/explore/state/utils.ts

```diff
@@ -30,19 +42,27 @@ export const storeGraphStyle = (graphStyle: string): void => {
   store.set(GRAPH_STYLE_KEY, graphStyle);
 };
 
-const LOGS_VOLUME_ENABLED_KEY = SETTINGS_KEYS.enableVolumeHistogram;
-export const storeLogsVolumeEnabled = (enabled: boolean): void => {
-  store.set(LOGS_VOLUME_ENABLED_KEY, enabled ? 'true' : 'false');
+export const storeSuppQueryEnabled = (enabled: boolean, type: SuppQueryType): void => {
+  if (suppQuerySettings[type]) {
+    store.set(suppQuerySettings[type], enabled ? 'true' : 'false');
+  }
 };
 
-const loadLogsVolumeEnabled = (): boolean => {
-  const data = store.get(LOGS_VOLUME_ENABLED_KEY);
-  // we default to `enabled=true`
-  if (data === 'false') {
-    return false;
-  }
+export const loadSuppQueries = (): SuppQueries => {
+  // We default to true for all supp queries
+  const suppQueries: SuppQueries = {
+    [SuppQueryType.LogsVolume]: { enabled: true },
+  };
 
-  return true;
+  for (const type of Object.keys(SuppQueryType)) {
```

Comment:
I think I see the reasoning behind this, but the more I look at it the less I think is a good approach. Here, you're reading keys from `SuppQueryType` only to check `Object.values(SuppQueryType).includes(type as SuppQueryType);` again, which it obviously is. I'll try to come up with an alternative approach.

Reply:
I guess using something like the following could be helpful

```typescript
export function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
  return Object.keys(obj) as K[];
}
```

and then use it in the loop like that:

```typescript
for (const type of enumKeys(SuppQueryType)) {
    // Only if "false" value in local storage, we disable it
    if (store.get(suppQuerySettings[type]) === 'false') {
      suppQueries[type] = { enabled: false };
    }
}
```

https://www.petermorlion.com/iterating-a-typescript-enum/

Reply:
> unnecessary dynamic programming

I think the purpose of this is to be dynamic, as `LogsVolume` won't be the only one. However I also don't think the hardcoded object in line 54 is optimal and should be dynamic.

Reply:
Even if dynamic, at some point we would need to specify/harcode all possible keys (siblings of `LogsVolume`). If we just need to do what `loadSuppQueries()` is doing in this PR, my advice would be to use the simplest implementation.

If we need to do extra work, or iterate over the keys again, I would think about adding an array of keys to use in the iteration (e.g. `const sup_query_types [ 'LogsVolume', '', ... ]`, and avoid doing `object.keys()` or `object.values()` to an enum. At least I don't find the produced code to be pleasant.

Reply:
Yeah, I was thinking about unifying  values of `suppQueryType.LogsVolume` and `SETTINGS_KEYS.enableVolumeHistogram` so it would be same and this would be removed. But as is this refactor, I didn't want to change it here.  What are your thoughts about this?

Reply:
Sounds good to me to change it as well.

Reply:
Updated to  https://github.com/grafana/grafana/pull/60998/commits/2c7b92ea0ec7a9f14049161befe6ad5ca8824f53 to use `const type of SUPPORTING_QUERY_TYPES`. Moreover, I have added comment to `supportingQuerySettings` that goal is to remove it and use the same values for settings and for enum.